### DISCUSSION TITLE
Negative lookahead for pageExtensions in NextJS Config

### DISCUSSION THREAD

**Discussion by stefanooni** [vercel/next.js#41244] [Help] [answered]
I am unsure how the array of `pageExtensions` is handled, but I assumed paths would be matched against a blob pattern, however it doesn't seem to be the case.

I have a bunch of pages with `.tsx` extension, however, I would like to exclude some pages which have the extension`.ignore.tsx`, however I cannot seem to get it to work. I have tried:

```
pageExtensions: ['(ignore.){0}tsx']
```

Any help would be appreciated.

**Comment by icyJoseph** [comment, marked as answer]
Could you try with: `pageExtensions: ["(?<!test.)(js(x?)|ts(x?))"]`, it uses a negative look behind assertion, which is not fully supported in browser environments, but it is supported in V8 engines, AFAIK.

**Reply by stefanooni** [reply]
Thank you, I was able to put this behind a ternary operator for checking if the page was being built or being served via development.
